src/material-experimental/mdc-menu/_menu-theme.scss

```diff
@@ -47,6 +48,18 @@
 @mixin typography($config-or-theme) {
   $config: typography.private-typography-to-2018-config(
       theming.get-typography-config($config-or-theme));
+
+  // When the 2014 typography config is mapped to the 2018 config, the font-weight that winds up
+  // getting used for menu items looks slightly off. In this case we override it to a value that
+  // looks better.
+  @if (typography.private-typography-is-2014-config($config-or-theme)) {
```

Comment:
Should we fix this at a higher level? I think that select and autocomplete (and maybe list?) use the same set of mixins.

Reply:
Yeah that's a good point. Let's merge this PR because its smaller in scope and will unblock me for the menu migrations I'm working on, but as a follow-up I'll try to address this more holistically

Reply:
So @andrewseguin made the opposite hack for table when he was working on that migration. There the font wasn't bold *enough*. Looking into it more I found that if we just ignore how the levels are named, they actually map better visually if we just swap them, so that's what I've done in https://github.com/angular/components/pull/23618 which allows us to remove both this hack and Andrew's 🎉